runtime-sdk/src/dispatcher.rs

```diff
@@ -365,8 +366,19 @@ impl<R: Runtime> Dispatcher<R> {
                     Ok(cbor::to_value(block_weight_limits))
                 }
                 // Runtime methods.
-                _ => R::Modules::dispatch_query(ctx, method, args)
-                    .ok_or_else(|| modules::core::Error::InvalidMethod(method.into()))?,
+                _ => {
+                    // Even though queries are not allowed to perform any private key ops, we
+                    // reduce the attack surface by completely disallowing them in secure builds.
+                    //
+                    // If one needs to perform runtime queries, she should use a regular build of
+                    // the runtime for that.
+                    if BUILD_INFO.is_secure {
+                        return Err(modules::core::Error::InvalidMethod(method.into()).into());
```

Comment:
might be nice to have a separate error message for this condition. but I guess it's not critical for this

Reply:
Agreed, let's add a separate variant.